internal/datastreams/processor.go

```diff
@@ -468,8 +472,10 @@ func (p *Processor) SetCheckpointWithParams(ctx context.Context, params options.
 	if params.ServiceOverride != "" {
 		service = params.ServiceOverride
 	}
+	processTags := processtags.GlobalTags().Slice()
```

Comment:
how expensive is this operation? Is there a copy involved here? This method is called a lot of times

Reply:
there is no copy, basically the content of this method is:

```go
func (p *ProcessTags) Slice() []string {
    if p == nil {
        return nil
    }
    p.mu.RLock()
    defer p.mu.RUnlock()
    return p.slice
}
```